src/TokenizerFwf.cpp

```diff
@@ -164,6 +171,14 @@ Token TokenizerFwf::nextToken() {
     row_++;
     col_ = 0;
 
+    // Proceed to the end of the line. This is needed in case the last column
+    // in the file is not being read.
+    while(fieldEnd != end_ && *fieldEnd != '\r' && *fieldEnd != '\n') {
```

Comment:
Yes there is room for improvement.
1. You don't have to proceed to the end of line if you are short. `tooShort = true` (you are already there).
2. You don't have to proceed to the end of line if format is ragged (reading the ragged column will do this).
3. However if not short and not ragged then you will not be at the EOL if the user does not wish to read the final column. In that case you have to proceed to EOL and you don't know the width, therefore the while loop.